code/modules/vending/clothingbooth.dm

```diff
@@ -27,16 +27,13 @@ var/list/clothingbooth_items = list()
 /obj/machinery/clothingbooth/proc/uisetup(var/mob/user)
 	if(!user.client)
 		return
-	//if(isnull(user.client.preferences.preview_icon))
+	if(!ishuman(user))
+		return
 	user << browse_rsc('browserassets/css/clothingbooth.css')
 	user << browse_rsc('browserassets/js/clothingbooth.js')
-	var/mob/living/carbon/human/H = user
-	if(istype(H) && H.flat_icon)
-		user << browse_rsc(H.flat_icon, "previewimage.png")
-	else
-		user.client.preferences.AH = user.bioHolder.mobAppearance
-		user.client.preferences.update_preview_icon()
-		user << browse_rsc(user.client.preferences.preview_icon, "previewimage.png")
+	user.client.preferences.AH = user.bioHolder.mobAppearance
+	user.client.preferences.update_preview_icon()
+	user << browse_rsc(user.client.preferences.preview_icon, "previewimage.png")
```

Comment:
This will always cause it to use your first saved character instead of the one you are using currently.

Reply:
i was thinking about replacing it with a dummy player sprite mannequin for now actually.

Reply:
That sounds like the best solution probably!

Reply:
as it is, it's better than being your last built flat icon that's facing left and wearing a bunch of clothes :P